src/main/java/org/openrewrite/java/spring/security6/UpdateRequestCache.java

```diff
@@ -0,0 +1,177 @@
+    );
+
+    private static final MethodMatcher REQUEST_CACHE_MATCHER = new MethodMatcher(
+        "org.springframework.security.config.annotation.web.configurers.RequestCacheConfigurer requestCache(org.springframework.security.web.savedrequest.RequestCache)"
+    );
+
+    @Override
+    public String getDisplayName() {
+        return "Keep the default RequestCache querying behavior in Spring Security 5";
+    }
+
+    @Override
+    public String getDescription() {
+        return "By default, Spring Security 5 queries the saved request on every request, which means that in a " +
+               "typical setup, the HttpSession is queried on every request to use the RequestCache. In Spring " +
+               "Security 6, the default behavior has changed, and RequestCache will only be queried for a cached " +
+               "request if the HTTP parameter \"continue\" is defined. To maintain the same default behavior as " +
+               "Spring Security 5, either explicitly add the HTTP parameter \"continue\" to every request or use " +
+               "NullRequestCache to override the default behavior.";
+    }
+
+    @Override
+    public @Nullable Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(5);
+    }
+
+    @Override
+    protected @Nullable TreeVisitor<?, ExecutionContext> getSingleSourceApplicableTest() {
+        return new UsesMethod<>(REQUEST_CACHE_MATCHER);
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new JavaIsoVisitor<ExecutionContext>() {
+
+            @Override
+            public J.Block visitBlock(J.Block block, ExecutionContext executionContext) {
+                block = super.visitBlock(block, executionContext);
+                List<Statement> statements = block.getStatements();
+
+                boolean hasContinueParameterStatement = findContinueParameterStatement(statements);
+                if (hasContinueParameterStatement) {
+                    return block;
+                }
+
+                for (int i = 0; i < statements.size(); i++) {
+                    Statement statement = statements.get(i);
+                    if (isNewHttpSessionRequestCacheStatement(statement)) {
+                        JavaTemplate template = JavaTemplate.builder(this::getCursor,
+                                "#{any()}.setMatchingRequestParameterName(\"continue\");")
+                            .javaParser(JavaParser.fromJavaVersion()
+                                .classpath("spring-security-web"))
+                            .imports(
+                                "org.springframework.security.web.savedrequest.HttpSessionRequestCache")
+                            .build();
+
+                        statement = statement.withTemplate(template, statement.getCoordinates().replace(), getSelect(statement));
+                        statements.add(i + 1, statement);
```

Comment:
Never directly mutate a collection of AST elements. Ideally the collections themselves would be immutable. 
Our change detection and diff calculation can be messed up by this sort of mutation. To add an element use `ListUtils.concat()`. 

Even if it does not cause a problem in this specific case, we don't want examples of this mistake floating around for people to copy from

Reply:
Yeah, However, I see there is currently no method in `ListUtils` to support inserting an element into a specific index. I think we need to add two methods to `ListUtils`
1. add an element to a specific index
2. remove an element from a specific index

Reply:
I see an example that returning `null` for a `visitMethodDeclaration` in the recipe `RemoveUnusedPrivateMethods` can remove that method, so for the statement removing, I think I can do the similar without a method from ListUtils

Reply:
Added a insert method to `ListUtils` and changed to call that method instead here